src/components/Form.js

```diff
@@ -1,35 +1,74 @@
 import React from 'react'
 import styled from 'styled-components/macro'
 import { NavLink, useHistory } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
+import { v4 as uuidv4 } from 'uuid'
 
 export default function Form({ onSubmit }) {
   const history = useHistory()
+  const { register, handleSubmit, errors } = useForm()
+  const onFormSubmit = (newReview) => {
+    onSubmit({ ...newReview, id: uuidv4() })
+    history.push('/')
+  }
   return (
-    <StyledForm onSubmit={handleSubmit}>
+    <StyledForm onSubmit={handleSubmit(onFormSubmit)}>
       <label>
         Title
-        <Input autoFocus name="title" />
+        <Input autoFocus name="title" ref={register({ required: true })} />
       </label>
+      {errors.title && errors.title.type === 'required' && (
+        <StyledErrorMessage>Title is required!</StyledErrorMessage>
+      )}
+
       <label>
-        Rating (0.0 - 10.0)
-        <Input name="rating" />
+        Rating (1-100)
+        <Input
+          name="rating"
+          ref={register({ required: true, pattern: /^[1-9][0-9]?$|^100$/ })}
+        />
```

Comment:
You could also use native browser validation to prevent input of negative numbers or text characters.
```suggestion
        <Input
          name="rating"
          type="number"
          min=1
          max=100
          ref={register({ required: true, pattern: /^[1-9][0-9]?$|^100$/ })}
        />
```

Reply:
I think this is already prevented, because of the pattern. Or is there a extra benefit with the native browser validation I don't know about?